apps/wallet/src/ui/app/staking/validator-detail/index.tsx

```diff
@@ -0,0 +1,264 @@
+        ({ validatorAddress }) => validatorAddress === validatorAddressParams
+    );
+
+    const validatorName = validatorData?.name || 'Loading...';
+
+    // TODO: get logo from validator data
+    const logo = null;
+
+    return (
+        <Overlay
+            showModal={showModal}
+            setShowModal={setShowModal}
+            title={
+                <div className="flex gap-2 items-center capitalize">
+                    <ImageIcon src={logo} alt={validatorName} size="small" />
+
+                    {validatorName}
+                </div>
+            }
+            closeIcon={SuiIcons.Close}
+            closeOverlay={close}
+        >
+            <Loading
+                className="w-full flex justify-center items-center"
+                loading={loading || pendingDelegationsLoading}
+            >
+                {showError && error && (
+                    <Alert className="mb-2">
+                        <strong>Sync error (data might be outdated).</strong>
+                        <small>{error.message}</small>
+                    </Alert>
+                )}
+
+                {validatorData && (
+                    <ValidatorDetailCard
+                        validatorAddress={validatorData.validatorAddress}
+                        pendingDelegationAmount={validatorData.staked || 0n}
+                        suiEarned={0n}
+                        apy={'0.00%'}
+                        commissionRate={0}
+                    />
+                )}
+            </Loading>
+        </Overlay>
+    );
+}
+
+type ValidatorDetailCardProps = {
+    validatorAddress: string;
+    pendingDelegationAmount: bigint;
+    suiEarned: bigint;
+    apy: number | string;
+    commissionRate: number;
+};
+
+function ValidatorDetailCard({
+    validatorAddress,
+    pendingDelegationAmount,
+    suiEarned,
+    apy,
+    commissionRate,
+}: ValidatorDetailCardProps) {
+    const totalStaked = useAppSelector(totalActiveStakedSelector);
+    const pendingStake = pendingDelegationAmount || 0n;
+    const totalStakedIncludingPending = totalStaked + pendingStake;
+
+    const stakeByValidatorAddress = `/stake/new?address=${encodeURIComponent(
+        validatorAddress
+    )}`;
+
+    const cardHeader = (
+        <div className="grid grid-cols-2 divide-x divide-solid divide-gray-45 divide-y-0 w-full">
+            <CardItem
+                title="Your Stake"
+                value={
+                    <CoinBalance
+                        balance={totalStakedIncludingPending}
+                        type={GAS_TYPE_ARG}
+                        diffSymbol
+                    />
+                }
+            />
+
+            <CardItem
+                title="EARNED"
+                value={
+                    <CoinBalance
+                        balance={suiEarned}
+                        type={GAS_TYPE_ARG}
+                        mode="neutral"
+                        className="!text-gray-60"
```

Comment:
We should improve these component APIs so that you don't need to do this. Not blocking for merge though.